examples/common/load_dds.adb

```diff
@@ -0,0 +1,277 @@
+
+with System;
+with Interfaces;
+
+with Ada.Exceptions; use Ada.Exceptions;
+with Ada.IO_Exceptions; use Ada.IO_Exceptions;
+with Ada.Text_IO; use Ada.Text_IO;
+with Ada.Unchecked_Conversion;
+
+with GL.Types;
+with GL.Low_Level.Enums;
+with GL.Objects.Buffers;
+with GL.Objects.Textures.Targets;
+with GL.Objects.Textures.With_2D_Loader;
+with GL.Pixels;
+
+with IO;
+
+--  Extracted from SOIL.adb  Direct_Load_DDS
+procedure Load_DDS (File_Name  : String;
+                        theTexture : out GL.Objects.Textures.Texture) is
```

Comment:
I've gotten rid of all Unchecked_Conversions except for Byte_To_Character as I can't figure out how to otherwise convert a byte to a character which (I thought) should be simple.
As for translating to Ada; two reasons:
1. www.opengl-tutorial.org/beginners-tutorials/tutorial-5-a-textured-cube provides its own loadDDS procedure.
2. My experience with SOIL has been very frustrating. When it fails it gives no real indication of why it fails, only: "raised SOIL.SOIL_ERROR : Unable to open file".
This one was for a png file which I checked exists using  Ada.Directories.Exists.
What is more annoying is that SOIL has been able to read a png file in another example.
I've never had any success with getting SOIL to read DDS files.
I've even tried using the GNAT GPS debugger but have been unable to track the code to the problem area.
I've tried to figure out what might be wrong from the SOIL source but have had no success with that either.
Because of these problems I had started to convert all of SOIL to Ada but the task began to look overwhelming from my point of view requiring more effort than I am prepared to allocate.
In contrast, coding the DDS loader, based on SOIL's DDS loader was not too difficult particularly with the help of the DDS specification and Hex-Editor to inspect an actual DDS file.
However, I don't think I'm prepared to do the same for png files.
Any advice on tracking down the png loading problem will be much appreciated.